servicetalk-grpc-api/src/main/java/io/servicetalk/grpc/api/GrpcStatusCode.java

```diff
@@ -63,18 +61,31 @@
     /** Cannot authenticate the client. */
     UNAUTHENTICATED(16);
 
-    private static final Map<Integer, GrpcStatusCode> INT_TO_STATUS_CODE_MAP;
+    private static final GrpcStatusCode[] INT_TO_STATUS_CODE_MAP;
+    private static final GrpcStatusCode[] H2_ERROR_TO_STATUS_CODE_MAP;
 
     static {
-        final Map<Integer, GrpcStatusCode> intToStatusCodeMap = new HashMap<>(GrpcStatusCode.values().length);
-        for (GrpcStatusCode code : GrpcStatusCode.values()) {
-            GrpcStatusCode replaced = intToStatusCodeMap.put(code.value(), code);
-            if (replaced != null) {
-                throw new IllegalStateException(String.format("GrpcStatusCode value %d used by both %s and %s",
-                        code.value(), replaced, code));
-            }
+        final GrpcStatusCode[] statusCodes = GrpcStatusCode.values();
```

Comment:
As part of API stability, we should consider using the final class approach for `GrpcStatusCode` either. The code is in integer range. There is a change a new codes may be added in newer gRPC protocol versions.

Reply:
I think enum is OK in this case bcz `UNKNOWN` accounts for this.

https://github.com/grpc/grpc-java/blob/master/api/src/main/java/io/grpc/Status.java#L73-L80
```java
/**
     * Unknown error.  An example of where this error may be returned is
     * if a Status value received from another address space belongs to
     * an error-space that is not known in this address space.  Also
     * errors raised by APIs that do not return enough error information
     * may be converted to this error.
     */
    UNKNOWN(2),
```

Reply:
I'm less concerned with inability to map to something, more with something like gRPC2 is released in future with 10 more status codes defined. Adding new enum values may be a breaking change for users.

Reply:
yip fair enough, we can add it to the list. also note that gRPC-java represent this as an enum.

Reply:
I saw that, but they can release grpc-java-2, for us it will be harder